src/main/java/frc/robot/Apriltags.java

```diff
@@ -0,0 +1,164 @@
+       
+           public Apriltags()
+           {
+               table = NetworkTableInstance.getDefault().getTable("limelight");
+           }
+
+           public Apriltags(String name) {
+               table = NetworkTableInstance.getDefault().getTable(name);
+           }
+       
+           public Apriltags(NetworkTable table) {
+               this.table = table;
+           }
+
+   
+           /**
+            * @return whether or not the limelight has any valid targets
+            */
+           public boolean isTargetFound() {
+
+               return table.getEntry("tv").getDouble(0) == 0f; 
+               // changed this to "not" as it was returning opposite
+           }
+       
+           /**
+            * @return the horizontal offset from crosshair to target (-29.8 to +29.8 degrees)
+            */
+           public double getHorizontalDegToTarget() {
+            double horizontalDegOffset = table.getEntry("tx").getDouble(0);
+
+            return horizontalDegOffset;
+           }
+       
+           /**
+            * @return vertical offset from crosshair to target (-24.85 to +24.85 degrees)
+            */
+           public double getVerticalDegToTarget() {
+            double verticalDegOffset = table.getEntry("ty").getDouble(0);
+            
+            return verticalDegOffset;
+           }
+       
+           /**
+            * @param pipeline sets the limelight’s current pipeline
+            */
+           public void setPipeline(int pipeline) {
+               if (pipeline < 0) {
+                   pipeline = 1;
+                   throw new IllegalArgumentException("Pipeline can not be less than zero");
+               } else if(pipeline > 9) {
+                   pipeline = 2;
+                   throw new IllegalArgumentException("Pipeline can not be greater than one");
+               }
+               table.getEntry("pipeline").setValue(pipeline);
+               table.getEntry("pipeline").setNumber(pipeline);
+           }
+       
+           /**
+            * @return current pipeline of limelight
+            */
+           public double getPipeline() {
+               return table.getEntry("pipeline").getDouble(0.0);
+           }
+       
+           /**
+            * @return distance from target in inches
+           */
+           public double estimateHorizontalDistance(){
+               double veritcalTargetAngle = getVerticalDegToTarget();
+               double height = kTargetHeight - kLensHeight;
+               double angleInRadians = (kMountAngle + veritcalTargetAngle) * (Math.PI / 180);
+
+               double horizontalDistance = height / Math.tan(angleInRadians);
+               //Determines the distance with camera height minus target height devided by tangent of camera angle plus the vertical offset angle
+               return horizontalDistance;
+           }
+       
+           public double estimateVerticalDistance(double horizontalDistance){
```

Comment:
Nomenclature wise, changing this to horizontal distance would be great :) 

```suggestion
           public double estimateHorizontalDistance(double horizontalDistance){
```